core/plugins/org.polarsys.capella.core.projection.scenario/src/org/polarsys/capella/core/projection/scenario/commands/ESToISCommand.java

```diff
@@ -83,34 +81,45 @@ protected AbstractTransform getTransformation(EObject element_p) {
       EObject eObject = it.next();
 
       if (eObject instanceof Scenario) {
-        Scenario scenario = (Scenario) eObject;
-        if (postRun || isScenarioValid(scenario)) {
-          // add also the scenario referenced by this scenario
-          Set<EObject> referencesScenarios = new HashSet<EObject>();
-          searchReferencedScenarios(scenario, referencesScenarios, result);
-          result.addAll(referencesScenarios);
-          result.add(scenario);
-        }
+        addAllScenarios((Scenario) eObject, result, postRun);
         it.prune();
       }
       // TODO we can prune many objects to limit lookup
     }
     return result;
   }
   
-/*
- * recursively, search for all referenced scenarios
- */
-  protected void searchReferencedScenarios(Scenario scenario, Set<EObject> references, Collection<EObject> analizedScenarios) {
-    if(analizedScenarios.contains(scenario))
-       return;
+  private void addAllScenarios(Scenario scenario, Collection<EObject> result, boolean postRun) {
+    if (postRun || isScenarioValid(scenario)) {
+      // add also the scenario referenced by this scenario
+      Set<EObject> referencesScenarios = new HashSet<EObject>();
+      searchReferencedScenarios(scenario, referencesScenarios, result, postRun);
+      result.addAll(referencesScenarios);
+      result.add(scenario);
+    }
+  }
+  
+  /*
+   * recursively, search for all referenced scenarios
+   */
+  protected void searchReferencedScenarios(Scenario scenario, Set<EObject> references,
+      Collection<EObject> analizedScenarios, boolean postRun) {
```

Comment:
analizedScenarios is always empty no ? it doesnt seems to me modified

Reply:
is not empty always, in the case we select multiple scenarios  (when in search scenarios goes trough the scenarios).

Reply:
Its hard to understand the code and what is difference between references, analizedScenarios.
can you simplify a bit.
Thanks

Reply:
ok